includes/utils.php

```diff
@@ -779,4 +779,92 @@ public static function is_elementor_path( $path ) {
 				return false !== strpos( $path, $p );
 			} );
 	}
+
+	/**
+	 * Get the correct locale for WPML, Polylang or WordPress.
+	 *
+	 * @since 3.7.0
+	 *
+	 * @return string
+	 */
+	public static function get_current_locale() {
+		if ( defined( 'POLYLANG_VERSION' ) ) {
+			return self::get_polylang_post_locale();
+		} elseif ( defined( 'ICL_SITEPRESS_VERSION' ) ) {
+			return self::get_wpml_post_locale();
+		} else {
+			return get_locale();
+		}
+	}
+
+	/**
+	 * Get Polylang post locale.
+	 *
+	 * @since 3.7.0
+	 *
+	 * @return string
+	 */
+	public static function get_polylang_post_locale() {
+		global $polylang;
+
+		$post_id = filter_input( INPUT_POST, 'editor_post_id' );
+
+		return $polylang->model->post->get_language( $post_id, 'locale' )->locale;
+	}
+
+	/**
+	 * Get WPML post locale.
+	 *
+	 * @since 3.7.0
+	 *
+	 * @return string
+	 */
+	public static function get_wpml_post_locale() {
+		$post_id = filter_input( INPUT_POST, 'editor_post_id' );
+
+		return apply_filters( 'wpml_post_language_details', null, $post_id )['locale'];
```

Comment:
```suggestion
		$wpml_post_language_details =  null;

		/**
		 * Post language details for WPML plugin.
		 *
		 * @since 3.7.0
		 *
		 * @param array|null $wpml_post_language_details
		 * @param int        $post_id
		 */
		$wpml_post_language_details = apply_filters( 'wpml_post_language_details', $wpml_post_language_details, $post_id );

		return $wpml_post_language_details['locale'];
```

Use the filters best practice. Define a variable, filter the result, return the variable.